src/main/java/paymelah/model/debt/Debt.java

```diff
@@ -79,7 +79,17 @@ public Debt setPaid(boolean isPaid) {
      */
     public int compareDateTimeWith(Debt o) {
         int dateComparison = this.date.compareTo(o.date);
-        return dateComparison == 0 ? this.time.compareTo(o.time) : dateComparison;
+        int timeComparison = this.time.compareTo(o.time);
+        return dateComparison == 0
+                ? timeComparison == 0 ? compareDescriptionWith(o) : timeComparison
```

Comment:
What if the descriptions are the same?

Reply:
If description same then sort by money? :')))